spec/controllers/submissions_controller_spec.rb

```diff
@@ -209,4 +255,151 @@
       it_behaves_like "edit_failure"
     end
   end
+
+  describe "#downloadAll" do
+    include_context "controllers shared context"
+    context "when user is Instructor of class with submissions" do
+      let!(:user) { instructor_user }
+      it "downloads all submissions for an assessmetn" do
+        sign_in(user)
+        get :downloadAll, params: { course_name: @course.name, assessment_name: @assessment.name }
+        expect(response).to be_successful
+        File.binwrite("tmp/test2.zip", response.parsed_body)
+        Zip::File.open("tmp/test2.zip") do |zip_file|
+          zip_file.each do |entry|
+            if entry.file?
+              content = entry.get_input_stream.read
+              expect(content).to match(/Hello Dave/m)
+            end
+          end
+        end
+        File.delete("tmp/test2.zip")
+      end
+    end
+  end
+
+  describe "#download" do
+    include_context "controllers shared context"
+    context "when user is Instructor of class with submissions" do
+      let!(:user) { instructor_user }
+      it "downloads a student's submission" do
+        sign_in(user)
+        get :download, params: { course_name: @course.name, assessment_name: @assessment.name,
+                                 id: get_first_submission_by_assessment(@assessment).id }
+        expect(response).to be_successful
+        File.binwrite("tmp/test.c", response.parsed_body)
+        File.open("tmp/test.c") do |file|
+          file_data = file.read
+          expect(file_data).to match(/Hello Dave/m)
+        end
+      end
+    end
+    context "when user is student and downloads own submission" do
+      let!(:user) { student_user }
+      it "downloads submission" do
+        sign_in(user)
+        submission = Submission.where(course_user_datum_id: get_first_cud_by_uid(user.id)).first
+        get :download, params: { course_name: @course.name, assessment_name: @assessment.name,
+                                 id: submission.id }
+        expect(response).to be_successful
+        File.binwrite("tmp/test.c", response.parsed_body)
+        File.open("tmp/test.c") do |file|
+          file_data = file.read
+          expect(file_data).to match(/Hello Dave/m)
+        end
+      end
+    end
+  end
+
+  describe "#missing" do
+    include_context "controllers shared context"
+    context "when user is student" do
+      let!(:user) { student_user }
+      it "gets missing submissions" do
```

Comment:
maybe I'm understanding wrong but should this be doesn't get missing submissions bc it's expecting the response to fail

Reply:
Indeed, the name is inaccurate.